src/migrations/20230802000000-internal-goal-duplicate-cleanup.js

```diff
@@ -0,0 +1,2438 @@
+            FULL OUTER JOIN "Objectives" oj ON g."id" = oj."goalId"
+            FULL OUTER JOIN "ObjectiveFiles" ojf ON oj."id" = ojf."objectiveId"
+            FULL OUTER JOIN "ObjectiveResources" ojr ON oj."id" = ojr."objectiveId"
+            FULL OUTER JOIN "ObjectiveTopics" ojt ON oj."id" = ojt."objectiveId"
+            FULL OUTER JOIN "ActivityReportObjectives" aro ON oj."id" = aro."objectiveId"
+            FULL OUTER JOIN "ActivityReportObjectiveFiles" arof ON aro."id" = arof."activityReportObjectiveId"
+            FULL OUTER JOIN "ActivityReportObjectiveResources" aror ON aro."id" = aror."activityReportObjectiveId"
+            FULL OUTER JOIN "ActivityReportObjectiveTopics" arot ON aro."id" = arot."activityReportObjectiveId"
+            GROUP BY gr."regionId"
+        );
+        INSERT INTO "PreCountStatsByRegion"
+        SELECT
+            -1 "regionId",
+            SUM("GoalsTotal"),
+            SUM("ActivityReportGoalsTotal"),
+            SUM("ActivityReportGoalResourcesTotal"),
+            SUM("ObjectivesTotal"),
+            SUM("ObjectiveFilesTotal"),
+            SUM("ObjectiveResourcesTotal"),
+            SUM("ObjectiveTopicsTotal"),
+            SUM("ActivityReportObjectivesTotal"),
+            SUM("ActivityReportObjectiveFilesTotal"),
+            SUM("ActivityReportObjectiveResourcesTotal"),
+            SUM("ActivityReportObjectiveTopicsTotal")
+        FROM "PreCountStatsByRegion";
+        SELECT * FROM "PreCountStatsByRegion";
+
+        DROP TABLE IF EXISTS "AllSmashedGoals";
+        CREATE TEMP TABLE "AllSmashedGoals" AS (
+            SELECT
+                g.id "goalId",
+                REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(
+                REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(
+                        LOWER(g.name),
+                        'grantee(s?)', 'recipient\\1', 'gi'),
+                        '\\bDLLs?\\b', 'dual language learners', 'gi'),
+                        '\\bEHS?\\b', 'early head start', 'gi'),
+                        '\\bELOF?\\b', 'head start early learning outcomes framework', 'gi'),
+                        '\\bFEI?\\b', 'full enrollment initiative', 'gi'),
+                        '\\bHS\\b', 'head start', 'gi'),
+                        '\\bHSELOF\\b', 'head start early learning outcomes framework', 'gi'),
+                        '\\bOHS\\b', 'office of head start', 'gi'),
+                        '\\bPBC\\b', 'practice based coaching', 'gi'),
+                        '\\bPD\\b', 'professional development', 'gi'),
+                        '\\bSR\\b', 'school readiness', 'gi'),
+                        '\\bDLL(s?)\\b', 'recipient\\1', 'gi'),
+                        '[^a-z0-9]', '', 'gi'),
+                        '[ \\t\\r\\n]', '', 'gi') "smashedName",
+                TRIM(g.name) "cleanName"
+            FROM "Goals" g
```

Comment:
Super unnecessary to do this on a dataset this small, but you can save expensive REGEX operations by doing
```
WITH unique_goal_names AS (
  SELECT DISTINCT name FROM "Goals"
),
smashed_names AS (
SELECT
  name,
  [regexes] "smashedName"
FROM unique_goal_names
)
SELECT
  g.id "goalId",
  "smashedName"
FROM "Goals" g
JOIN smashed_names sn
  ON g.name = sn.name
;
```